backend/src/game/dice_handler.py

```diff
@@ -0,0 +1,29 @@
+import backend.src.game.dice.dice as dice
+
+
+class DiceHandler:
+    def __init__(self):
+        self.dice_values = []
+        self.re_rolls_left = 0
+
+    def start_turn(self, starting_dice_count, re_roll_count):
+        self.dice_values = dice.roll_many(starting_dice_count)
+        self.re_rolls_left = re_roll_count
+
+    def re_roll_dice(self, indexes_of_dice_to_re_roll):
+        if self.re_rolls_left <= 0:
+            return
+
+        try:
+            for val in indexes_of_dice_to_re_roll:
+                self.dice_values[val] = dice.roll()
+            self.re_rolls_left -= 1
+
+        except IndexError:
```

Comment:
This index error only needs to wrap line 19, it does not have to be this far away

Reply:
I think it needs to wrap line 20 as well. I don't want to cost a turn if we don't actually get a valid result. Was this comment more about the extra line space I have on line 21? Cause that seems wrong